fe/fe-core/src/main/java/com/starrocks/analysis/CreateFunctionStmt.java

```diff
@@ -80,20 +86,129 @@
+        }
+
+        private void checkReturnJavaType(Method method, Class expType) throws AnalysisException {
+            checkJavaType(method, expType, method.getReturnType(), RETURN_FIELD_NAME);
+        }
 
+        private void checkJavaType(Method method, Class expType, Class ptype, String pname)
+                throws AnalysisException {
+            if (!expType.equals(ptype)) {
+                throw new AnalysisException(
+                        String.format("UDF class '%s' method '%s' parameter %s[%s] expect type %s",
+                                clazz.getCanonicalName(), method.getName(), pname, ptype.getCanonicalName(),
+                                expType.getCanonicalName()));
+            }
+        }
+
+        private void checkArgumentCount(Method method, int argumentCount)
+                throws AnalysisException {
+            if (method.getParameters().length != argumentCount) {
+                throw new AnalysisException(
+                        String.format("UDF class '%s' method '%s' expect argument count %d",
+                                clazz.getCanonicalName(), method.getName(), argumentCount));
+            }
+        }
+
+        private void checkParamUdfType(Method method, Type expType, Parameter p) throws AnalysisException {
+            checkUdfType(method, expType, p.getType(), p.getName());
+        }
+
+        private void checkReturnUdfType(Method method, Type expType) throws AnalysisException {
+            checkUdfType(method, expType, method.getReturnType(), RETURN_FIELD_NAME);
+        }
+
+        private void checkUdfType(Method method, Type expType, Class ptype, String pname)
+                throws AnalysisException {
+            if (!(expType instanceof ScalarType)) {
+                throw new AnalysisException(
+                        String.format("UDF class '%s' method '%s' does not support non-scalar type '%s'",
+                                clazz.getCanonicalName(), method.getName(), expType));
+            }
+            ScalarType scalarType = (ScalarType) expType;
+            Class cls = PrimitiveTypeToJavaClassType.get(scalarType.getPrimitiveType());
+            if (cls == null) {
+                throw new AnalysisException(
+                        String.format("UDF class '%s' method '%s' does not support type '%s'",
+                                clazz.getCanonicalName(), method.getName(), scalarType));
+            }
+            if (!cls.equals(ptype)) {
+                throw new AnalysisException(
+                        String.format("UDF class '%s' method '%s' parameter %s[%s] type does not match %s",
+                                clazz.getCanonicalName(), method.getName(), pname, ptype.getCanonicalName(),
+                                cls.getCanonicalName()));
+            }
+        }
+    }
+
+    private UDFInternalClass udfClass;
+    private UDFInternalClass udfStateClass;
```

Comment:
```suggestion
    private UDFInternalClass udafClass;
```

Reply:
`udfStateClass` is internal state class.